app/uk/gov/hmrc/selfservicetimetopay/controllers/CalculatorController.scala

```diff
@@ -97,11 +97,11 @@ class CalculatorController(calculatorConnector: CalculatorConnector) extends Tim
     sessionCache.get.flatMap {
       case Some(ttpData@TTPSubmission(Some(schedule), _, _, Some(Taxpayer(_, _, Some(sa))), _, _, CalculatorInput(debits, paymentToday, _, _, _, _), _)) =>
         val form = CalculatorForm.createPaymentTodayForm(debits.map(_.amount).sum).fill(paymentToday)
-        Future.successful(Ok(calculate_instalments_form(schedule,
+        Future.successful(Ok(calculate_instalments_form(schedule, Some(sa.debits),
```

Comment:
You should use debits from CalculatorInput. This will always be a Seq of debits or it will be empty. In the case it is empty this case won't catch it, this is handled as an error case, as you should not be allowed to navigate to this page with an empty sequence. If the user has logged in, then the debits in CalculatorInput will equal those stored in the sa section of taxpayer. As such, the debits in this case will always contain the debits you need.